stablehlo/dialect/Base.h

```diff
@@ -110,11 +110,23 @@ LogicalResult inferMostSpecificTypeComponents(
     std::optional<Location> location, TypeRange inputTypes,
     SmallVectorImpl<ShapedTypeComponents> &inferredReturnShapes);
 
+inline bool matchNumericType(DenseIntElementsAttr attr,
+                             SmallVector<int64_t> &result) {
+  // FIXME: Discuss if this should be signless 64?
```

Comment:
What is the intention of passing `matchInts(..., SmallVector<int64_t>)` vs `APInt`? Is this used to indicate signless integer type? Or is the intent that this would mean `i64` type? It is currently used as general signless integer type.

Here is a snippet from `DynamicConv`:

```
SmallVector<int64_t> padding;
if (failed(hlo::matchInts(op.getDPadding(), padding)))
  return rewriter.notifyMatchFailure(op, "expected constant d_padding");
```

DPadding is often i32 as well. Tests fail if `matchNumericType` were to check for `isSignlessInteger(64)` instead of just general signless integer.

Reply:
I've provided additional documentation around this in #1451. The overall idea was to have an int64_t-based API for when the underlying type doesn't matter (e.g. if we pass the results to int64_t-based APIs like ShapedTypeComponents), and an APInt-based API when it does matter (e.g. when we add the results together or something).